core/src/main/java/io/grpc/ForwardingChannelBuilder.java

```diff
@@ -145,6 +146,11 @@ public T maxInboundMessageSize(int max) {
     return thisT();
   }
 
+  @Override public T proxyDetector(ProxyDetector proxyDetector) {
```

Comment:
I don't think we'd want to pass in the ProxyDetector. It should either be a ProxySelector or a Proxy (and I'm not yet sure which; it seems the "power" of ProxySelector doesn't add much value, but still unclear).